content/projects/gsoc/2022/projects/pipeline-step-documentation-generator.adoc

```diff
@@ -15,30 +15,47 @@ mentors:
 - "arpoch"
 - "koushartasneem"
 links:
-  gitter: "jenkinsci/git-plugin"
-  draft: https://docs.google.com/document/d/1dHFEHbVPNyGRGtMF30df4O0UXoqgmEJpVVNoxxXYT1w/edit?usp=sharing
+  gitter: "jenkinsci/docs"
+  draft: https://docs.google.com/document/d/1gFA-EctmLyKWy_NfTfRL7cO-3mUjaS1aIcixyQ1ecBk/edit?usp=sharing
   idea: /projects/gsoc/2022/project-ideas/pipeline-step-documentation-generator
   meetings: "/projects/gsoc/2022/projects/pipeline-step-documentation-generator/#office-hours"
 ---
 
-//above links must be updated
-
-
 === Abstract
 
-(to be described)
+This project aims to improve the https://www.jenkins.io/doc/pipeline/steps/[Pipeline Steps Reference] that is generated automatically with the help of the plugin manager associated with a mock Jenkins instance. The improvements will lead to a smoother user experience and better usability of the generated documentation.
 
 === Rationale
 
-(to be described)
+Documenting the Pipeline steps is not an easy task. Currently, there are more than *600 plugins* that provide *1500+ steps* for use within a Pipeline script. Furthermore, these steps have several parameters that may be used along with them. All these steps are documented in the Asciidoc format, which is autogenerated in a specified format by a Java program. 
+
+Although the https://www.jenkins.io/doc/book/pipeline/getting-started/#snippet-generator[Pipeline Snippet Generator] is the best way to understand how a step should be used, there are two reasons why documenting all the steps is necessary. 
+
+. Not all users are aware of how the snippet generator works.
+. There must be a compilation of all the steps available across all the plugins.
+
+So, to work with documentation of this magnitude, it is required that we generate and format it so that the user can find the piece of information they are seeking effectively. Thousands of Pipeline developers all over the world will benefit from improvements in the same.
+
 
 === Implementation
-(to be described)
 
-=== Office hours
+The implementation revolves around two major changes.
+
+. Changes to `jenkins.io` - Change the navigation experience on the website. This includes changes in the layout as well as adding more navigation bars.
+. Changes to `pipeline-step-docs-generator` - Format the parameter details in a better way and add relevant information to clarify their usage.
+
+Apart from these changes, minor changes will be done to refactor the code and remove anomalies from the steps documentation. Also, the instructions regarding the snippet generator will be included in the documentation template to promote its usage.
+
+=== Office Hours
 
-Details about the Office Hours (general or specific)
+* General: https://docs.google.com/document/d/1ygRZnVtoIvuEKpwNeF_oVRVCV5NKcZD1_HMtWlUZguo/edit[Docs Office Hours] - Thursdays @ *1800 UTC*
```

Comment:
You might also want to add our group's office hours on Tuesdays & our meeting notes if someone wants to comment on that.

Reply:
Yes, I have added that in the project links section, under the _GSoC Project Notes_ title, and hyperlinked the meet link in the project-specific office hours title.